src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -286,7 +286,8 @@ private IReadOnlyList<SemanticRange> CombineSemanticRanges(params IReadOnlyList<
             var request = await _languageServer.SendRequestAsync(LanguageServerConstants.RazorProvideSemanticTokensEndpoint, parameter);
             var csharpResponse = await request.Returning<ProvideSemanticTokensResponse>(cancellationToken);
 
-            if (csharpResponse is null || csharpResponse.HostDocumentSyncVersion != documentVersion)
+            if (csharpResponse is null ||
+                (csharpResponse.HostDocumentSyncVersion != null && csharpResponse.HostDocumentSyncVersion != documentVersion))
```

Comment:
So we don't want to return null if `csharpResponse.HostDocumentSyncVersion` is null?
When does that usually happen?

Reply:
To my knowledge that should only ever happen when VSCode sends us a "fake" response to `razor/provideSemanticTokens` because it doesn't current have the API's we need to give a real response.